File: app/src/main/java/com/app/cancellingretrofitrequests/java/networking/ApiService.java
package com.app.cancellingretrofitrequests.java.networking;


import com.app.cancellingretrofitrequests.java.model.SearchResultModel;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ApiService {
    public static String BASE_URL = "https://demo.dataverse.org/api/";

    @GET("search")
    Call<SearchResultModel> getResult(@Query("q") String query);
}

File: app/src/main/java/com/app/cancellingretrofitrequests/kotlin/networking/ApiService.kt
package com.app.cancellingretrofitrequests.kotlin.networking

import com.app.cancellingretrofitrequests.kotlin.model.SearchResultModel
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Query


interface ApiService {
    companion object {
        val BASE_URL: String = "https://demo.dataverse.org/api/"
    }

    @GET("search")
    suspend fun getResult(
            @Query("q") query: String?): Response<SearchResultModel?>
}